### PR TITLE
feat(modal): add i18nDismissLabel prop

### PR DESCRIPTION
**Pull request by shleewhite**
- [x] add i18nDismissLabel prop to ModalHeader
- [x] add i18n story
- [x] add tests for default and custom i18nDismissLabel
- [x] add example to website
- [x] add prop to ModalHeader prop table on website

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: packages/paste-core/components/modal/__tests__/index.spec.tsx [hunk 1]
```diff
@@ -96,6 +96,44 @@ const MockInitalFocusModal: React.FC = () => {
...
+      <Modal
+        data-testid="modal"
+        aria-busy="true"
+        id="a-new-id"
+        ariaLabelledby={modalHeadingID}
+        isOpen
+        onDismiss={handleCloseMock}
+        size="default"
+      >
+        <ModalHeader data-testid="modal-header" i18nDismissLabel="foo bar">
```
Comment (TheSisb) [inline packages/paste-core/components/modal/__tests__/index.spec.tsx:112]: thought (non-blocking): for the purpose of the test, I wonder if we can only render the `<ModalHeader>` to speed up the test.

File: packages/paste-core/components/modal/stories/index.stories.tsx [hunk 2]
```diff
@@ -448,6 +449,52 @@ HeaderContent.story = {
...
+export const i18nProp = (): React.ReactNode => {
+  const [isOpen, setIsOpen] = React.useState(true);
+  const handleOpen = (): void => setIsOpen(true);
+  const handleClose = (): void => setIsOpen(false);
+  const modalHeadingID = useUID();
+
+  return (
+    <div>
+      <Button variant="primary" onClick={handleOpen}>
+        Abrir modal
```
Comment (TheSisb) [inline packages/paste-core/components/modal/stories/index.stories.tsx:461]: praise: very nice!

**Review by TheSisb — APPROVED** [review]

File: packages/paste-core/components/modal/__tests__/index.spec.tsx (hunk 1 above)
Comment (shleewhite) [reply packages/paste-core/components/modal/__tests__/index.spec.tsx:112]: I tried only rendering the header, but it needs the `ModalContext.Provider` :/

Comment (TheSisb) [reply packages/paste-core/components/modal/__tests__/index.spec.tsx:112]: thought (non-blocking): Did you try `<Modal><ModalHeader ...></Modal>`? Essentially removing anything that can be removed from needing to render.

Comment (shleewhite) [reply packages/paste-core/components/modal/__tests__/index.spec.tsx]: Ok! I can do that.

**Review by gloriliale — APPROVED** [review]
Fantastic!

**Pull request merged**